src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -504,27 +506,31 @@ public static List<String> getJacocoAgentOptions() {
         return new Statement() {
             @Override public void evaluate() throws Throwable {
                 System.out.println("=== Starting " + description);
-                if (war == null) {
-                    war = findJenkinsWar();
-                }
-                if (home.get() != null) {
+                try {
+                    jenkinsOptions("--webroot=" + createTempDirectory("webroot"), "--pluginroot=" + createTempDirectory("pluginroot"));
+                    if (war == null) {
+                        war = findJenkinsWar();
+                    }
+                    if (home.get() != null) {
+                        try {
+                            base.evaluate();
+                        } finally {
+                            stopJenkins();
+                        }
+                        return;
+                    }
                     try {
+                        home.set(tmp.allocate());
+                        if (!prepareHomeLazily) {
+                            provision();
+                        }
                         base.evaluate();
                     } finally {
                         stopJenkins();
                     }
-                    return;
-                }
-                try {
-                    home.set(tmp.allocate());
-                    if (!prepareHomeLazily) {
-                        provision();
-                    }
-                    base.evaluate();
                 } finally {
-                    stopJenkins();
                     try {
-                        deprovision();
+                        tmp.dispose();
```

Comment:
May be confusing depending on whether or not you have whitespace displayed: now just deleting all temp dirs when the test completes, regardless of #610 mode. (The other code in `deprovision` did not seem important since a rule instance is only expected to be used once.)